test/integration/targets/pids/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+# Test code for the pid module
+# Copyright 2018, Saranya sridharan
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: "Installing the psutil module"
+  pip:
+    name: psutil
+
+- name: "Checking the empty result" 
+  pids: 
+    name: "blahblah"
+  register: emptypids
+
+- name: "Verify that the list of Process IDs (PIDs) returned is empty"
+  assert:
+    that:
+    - emptypids is not changed
+    - emptypids.pids == [] 
+
+- name: "Picking a random process name"
+  command: echo "some-long-random-process-name-{{ 99999999 | random }}"
+  register: random_name
+
+- name: "finding the 'sleep' binary"
+  command: which sleep
+  register: find_sleep
+
+- name: "copying 'sleep' binary"
+  copy:
+    src: "{{ find_sleep.stdout }}"
+    dest: "{{ role_path }}/tasks/{{ random_name.stdout }}"
+    mode: "777"
+
+- name: "Running the copy of 'sleep' binary"
+  command: "sh {{ role_path }}/tasks/files/obtainpid.sh '{{ role_path }}/tasks/{{ random_name.stdout }}' '{{ role_path }}/tasks/obtainpid.txt'"
+  async: 100
+  poll: 0
+
+- name: "Checking the process IDs (PIDs) of sleep binary" 
+  pids: 
+    name: "{{ random_name.stdout }}"
+  register: pids
+    
+- name: "Reading pid from the file"
+  slurp:
+    src:  "{{ role_path }}/tasks/obtainpid.txt"
+  register: newpid
+
+- name: "Verify that the Process IDs (PIDs) returned is not empty and also equal to the PIDs obtained in console"
+  assert:
+    that:
+    - "pids.pids | join(' ')  == newpid.content | b64decode | trim"
+    - "pids.pids | length > 0"
```

Comment:
Add a test using a substring of `random_name.stdout` to make sure only exact matches work.